libimage/events.go

```diff
@@ -18,6 +18,8 @@ const (
 	EventTypeUnknown EventType = iota
 	// EventTypeImagePull represents an image pull.
 	EventTypeImagePull
+	// EventTypeImagePullError represents an image pull failed.
+	EventTypeImagePullError
```

Comment:
Do we need that here as well OR should we let Podman deal with it?  I feel like Podman should handle it.

Reply:
I am fine with going that way, but thought this might not be a breaking change, since we are adding a new event and not forcing apps that are looking for pullevent as a success indicator.

Reply:
We can break here as Podman is the only user/caller of c/common and we're below 1.0.0.

What I meant by breaking change is that we cannot make this user visible.  So external users Podman reading the events should not notice a change.  So we need a new pull_error (or something like that) for `podman events` (and the backends).

Reply:
Assume some automation that is reading the logs/events for pulled images.  Now, pull errors would show up and cause false positives.

Reply:
We need to know that a pull happened, if we want to put this into Podman then we have to plumb this into tons of different places, and we would probably need to plumb it into buildah as well.